web/client/components/widgets/enhancers/dependenciesToExtent.js

```diff
@@ -22,70 +22,80 @@ const { set } = require('../../../utils/ImmutableUtils');
+                    if (!mapSync || isEmpty(layerInCommon)) {
+                        return Observable.of({});
+                    }
+                    let filterObjCollection = {};
+                    if (dependencies.quickFilters) {
+                        filterObjCollection = {...filterObjCollection, ...composeFilterObject(filterObj, dependencies.quickFilters, dependencies.options)};
+                    }
+                    if (dependencies.filter) {
+                        filterObjCollection = {...filterObjCollection, ...FilterUtils.composeAttributeFilters([filterObjCollection, dependencies.filter])};
+                    }
+                    const featureTypeName = layerInCommon.name;
+                    if (!isEmpty(filterObjCollection)) {
+                        const wfsGetFeature = FilterUtils.toOGCFilter(featureTypeName, filterObjCollection, "1.1.0");
+                        return wpsBounds(getLayerUrl(layerInCommon), {wfsGetFeature })
+                            .switchMap(response => {
+                                let json;
+                                let sw;
+                                let ne;
+                                xml2js.parseString(response.data, {explicitArray: false}, (ignore, result) => {
+                                    json = result["ows:BoundingBox"];
+                                    sw = json["ows:LowerCorner"].split(" ");
+                                    ne = json["ows:UpperCorner"].split(" ");
+                                });
+                                if (json["ows:LowerCorner"] === "0.0 0.0" && json["ows:UpperCorner"] === "-1.0 -1.0") {
+                                    return Observable.of({});
+                                }
+                                const bounds4326 = {
+                                    minx: parseFloat(sw[0]),
+                                    miny: parseFloat(sw[1]),
+                                    maxx: parseFloat(ne[0]),
+                                    maxy: parseFloat(ne[1])
+                                };
+                                const bounds3857 = CoordinatesUtils.reprojectBbox(bounds4326, "EPSG:4326", "EPSG:3857");
+                                return Observable.of({
+                                    bounds4326,
+                                    bounds: {
+                                        minx: parseFloat(bounds3857[0]),
+                                        miny: parseFloat(bounds3857[1]),
+                                        maxx: parseFloat(bounds3857[2]),
+                                        maxy: parseFloat(bounds3857[3])
+                                    }
+                                });
+                            })
+                            .startWith({loading: true})
+                            .catch((error) => Observable.of({ error: "error getting bounds from filter", details: error }));
```

Comment:
ending this stream with loading: false (and resetting zoomToExtent, but if we use some different registry, it may be not necessary)